accepted/00087-containerized-branch.md

```diff
@@ -0,0 +1,192 @@
+
+### Smart TFTP server
+
+Instead of `image-sync formula`, which is responsible to download, extract and verify of kernel, initrd and filesystem image in this RFC together with [RFC about cobbler sync](https://github.com/uyuni-project/uyuni-rfc/pull/44) I propose usage of wrapper around [Facebook's TFTP server](https://github.com/facebook/fbtftp) which will translate TFTP requests to HTTP requests and forward to proxy HTTP server. HTTP server is then configured that such requests are forwarded to Uyuni server through Squid cache. Images, due to revision number nature and including kernel and initrd, are never stale so after first download through WAN subsequent terminal boots will use cached image, thus increasing performance.
+
+### Image building changes
+
+This direct access to initrd and kernel needs changes in how image are built. Currently after image is built, Kiwi is instructed to prepare so called kiwi bundles, one tar ball with all parts together. There is a choice between collecting this tarball as is and extracting it on the Server, or stop Kiwi bundling and collect individual files.
+In this RFC I opt for collecting individual files for simple reason not to waste cycles on build host for creating tarball only to be immediately extracted on the Server and increasing load on it. Negative is that current Retail templates do not use image compression by default, so unbundled images are several times bigger then bundled (e.g. for POS_Image_JeOS7 it is around 400MB in bundle and 1,6GB unbundled). This can be remedied by using compression by default in Kiwi templates.
+
+### PXE Management
+[pxe_management]: #pxe_management
+
+Deployment workflow needs two kinds of PXE configuration files - `default` for first time, not yet deployed terminals and `01-MAC` for individual terminals already deployed at least once.
+
+`default` configurations are different for each `saltboot group` as this configuration contains `MASTER=<proxyFQDN>` entry to remove the need of custom DNS server and this entry is different for each proxy server.
+
+
+### Networking
+
+Managing networking on proxy server is out of scope of containers and left on customer. This means that, similarly as we have with Cobbler autoinstallation now, we will ask customers to set their DHCP environment for PXE booting elsewhere. This eliminates `Dhcp formula` requirement and also majority of `Branch Network formula`. `Dhcp formula` can still be maintained as customer can use that for their separate dhcp management.
+
+`salt` CNAME and `ftp` hostnames are replaced by crafting special PXE kernel command (see [PXE Management](#pxe_management) line and salt pillars (see [Saltboot group](#saltboot_group). We will use this to eliminate `Bind formula`.
+
+
+# Workflow
+[workflow]: #workflow
+
+TBD user story
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we **not** do this?
+
+  * Change how Retail is configured and done
```

Comment:
Will the old workflow stay available for existing users?

Reply:
Yes, old workflow will work the same